src/signal.rs

```diff
@@ -822,6 +830,40 @@ pub fn gen_mut<G, F>(gen_mut: G) -> GenMut<G, F>
 }
 
 
+/// Create a new `Signal` that calls the enclosing function on each iteration.
+///
+/// # Example
+///
+/// ```rust
+/// extern crate sample;
+///
+/// use sample::{signal, Signal};
+///
+/// fn main() {
+///     let mut f = [0.0];
+///     let mut signal = signal::gen_mut(move || {
+///         f[0] += 0.1;
+///         f
+///     });
+///     let func = |x: &[f64; 1]| {
+///         assert_eq!(*x, [0.1]);
+///     };
+///     let mut inspected = signal::inspect(signal, func);
+///     let out = inspected.next();
+///     assert_eq!(out, [0.1]);
+/// }
+/// ```
+pub fn inspect<S, Func>(signal: S, func: Func) -> Inspect<S, Func>
+    where S: Signal,
+          Func: FnMut(&<S as Signal>::Frame) -> (),
```

Comment:
It should be possible (and more idiomatic) to write the `<S as Signal>::Frame` type as `S::Frame` here and in the `Signal` impl for `Inspect`. Normally the angle bracket syntax is only required if there is some ambiguity or when referring to an associated type of an associated type.